autoload/go/fmt.vim

```diff
@@ -95,6 +97,9 @@ function! go#fmt#Format(withGoimport) abort
     " Restore our cursor/windows positions.
     call winrestview(l:curw)
   endif
+
+  " be smart and jump to the line the new statement was added/removed
+  call cursor(line('.') + diff_offset, 0)
```

Comment:
Ah! That would do it. I'm guessing the second argument to `cursor()` is the column number?